solidity/dashboard/src/services/tokens-page.service.js

```diff
@@ -374,11 +365,86 @@ const getGranteeDelegations = async (
     false
   )
 
-  return await getDelegations(
+  const [delegations, undelegations] = await getDelegations(
     allOperators,
     initializationPeriod,
     undelegationPeriod
   )
+
+  return [delegations, undelegations, granteeGrants]
+}
+
+export const getCopiedDelegations = async (
+  ownerOrGrantee,
+  grantIds,
+  initializationPeriod,
+  undelegationPeriod
+) => {
+  const {
+    stakingPortBackerContract,
+    grantContract,
+    stakingContract,
+  } = await ContractsLoaded
+
+  const copiedEvents = await stakingPortBackerContract.getPastEvents(
+    "StakeCopied",
+    {
+      fromBlock: CONTRACT_DEPLOY_BLOCK_NUMBER.stakingPortBackerContract,
+      filter: { owner: ownerOrGrantee },
+    }
+  )
+
+  const operatorsToCheck = copiedEvents.map(
+    (event) => event.returnValues.operator
+  )
+
+  // We only want operators for whom the delegation has not been paid back.
+  const operatorsOfPortBacker = await getOperatorsOfOwner(
+    stakingPortBackerContract.options.address,
+    operatorsToCheck
+  )
+
+  // Get operators from delegations created from a grant.
+  let tokenGrantStakingEvents = []
+  if (!isEmptyArray(grantIds)) {
+    tokenGrantStakingEvents = await grantContract.getPastEvents(
+      "TokenGrantStaked",
+      {
+        fromBlock: CONTRACT_DEPLOY_BLOCK_NUMBER.grantContract,
+        filter: { grantId: grantIds },
+      }
+    )
+  }
+
+  // Scan `OperatorStaked` event by operator(indexed param) to get authorizer and beneeficiary.
+  const operatorToDetails = (
+    await stakingContract.getPastEvents("OperatorStaked", {
+      fromBlock: CONTRACT_DEPLOY_BLOCK_NUMBER.stakingContract,
+      filter: { operator: operatorsOfPortBacker },
+    })
+  ).reduce((reducer, _) => {
+    reducer[_.returnValues.operator] = _.returnValues
+    reducer[_.returnValues.operator].isCopiedStake = true
+
+    return reducer
+  }, {})
```

Comment:
Can we move this part before 
```
  // Get operators from delegations created from a grant.
  let tokenGrantStakingEvents = []
  if (!isEmptyArray(grantIds)) {
    tokenGrantStakingEvents = await grantContract.getPastEvents(
      "TokenGrantStaked",
      {
        fromBlock: CONTRACT_DEPLOY_BLOCK_NUMBER.grantContract,
        filter: { grantId: grantIds },
      }
    )
  }
```
?

We would have pulling `tokenGrantStakingEvents` and looping over them next to each other.